### ISSUE TITLE
zLinux install errors

### ISSUE THREAD

**Issue by Simnbf** [zowe/zowe-cli-scs-plugin#46]
I have a zlinux server (rhel 7, s390x) running zowe and serving as an internal jenkins agent. We're trying to install the secure credentials plugin and facing various issues. I'm ssh'd into the server as root as a bit of background. 
(I've redacted a few internal stuff in the logs and output below)

I've followed the steps here, all done up to the `zowe plugins install` step
https://github.com/zowe/zowe-cli-scs-plugin/blob/master/README.md#software-requirements

Below is the output off the install command, you can see it complains about access to create a directory despite running as root. Googling the problem has returned similar problems for npm install commands, but the additional parameters they suggest are invalid for the `zowe plugins install ` command.

```
root@x DEV # zowe plugins install @zowe/secure-credential-store-for-zowe-cli@zowe-v1-lts
Plug-ins within the Imperative CLI Framework can legitimately gain
control of the zowe CLI application during the execution of every command.
Install 3rd party plug-ins at your own risk.
 
Imperative's plugin installation program handles @zowe peer dependencies.
You can safely ignore NPM warnings about missing @zowe peer dependencies.
 
Registry = https://redacted.com/nexus/content/groups/redacted-npm-read/
 
_______________________________________________________________
prebuild-install WARN install self signed certificate in certificate chain
gyp ERR! configure error
gyp ERR! stack Error: EACCES: permission denied, mkdir '/root/.zowe/plugins/installed/lib/node_modules/@zowe/secure-credential-store-for-zowe-cli/node_modules/keytar/build'
gyp ERR! System Linux 3.10.0-1160.31.1.el7.s390x
gyp ERR! command "/opt/nodejs/node-v14.17.3-linux-s390x/bin/node" "/opt/nodejs/node-v14.17.3-linux-s390x/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /root/.zowe/plugins/installed/lib/node_modules/@zowe/secure-credential-store-for-zowe-cli/node_modules/keytar
gyp ERR! node -v v14.17.3
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! keytar@7.7.0 build: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the keytar@7.7.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN @zowe/secure-credential-store-for-zowe-cli@4.1.5 requires a peer of @zowe/cli@^6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN @zowe/secure-credential-store-for-zowe-cli@4.1.5 requires a peer of @zowe/imperative@^4.0.0 but none is installed. You must install peer dependencies yourself.
 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! keytar@7.7.0 install: `prebuild-install || npm run build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the keytar@7.7.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
 
npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-07-30T12_13_19_913Z-debug.log
Command Error:
Install Failed

```

I've attached the full npm logs as well. I can succesfully create the directories myself, and set wide open permission bits, but the install command appears to delete them before creating them again.

[scs install.txt](https://github.com/zowe/zowe-cli-scs-plugin/files/6907604/scs.install.txt)

**Comment by skurnevich** [comment]
Hi Simon, so the CLI itself is working fine, only SCS plugin is not installed is that true? 

I see you are following the docs so have you tried to solve EACCES error as mentioned in [pt.3 - Notes section](https://docs.zowe.org/stable/user-guide/cli-installcli/#installing-zowe-cli-from-an-online-registry) and it didn't help ([npm EACCES](https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally))?

Please also consider to check out similar issues on zowe-cli:
This one about EACCES - https://github.com/zowe/zowe-cli/issues/827
Or this related to linux installations as root - https://github.com/zowe/zowe-cli/issues/666

**Comment by t1m0thyj** [comment]
> Googling the problem has returned similar problems for npm install commands, but the additional parameters they suggest are invalid for the zowe plugins install command.

Although it's not possible to pass the `unsafe-perm` flag on the zowe plugins install command, you can globally set it in NPM config:
```npm config set unsafe-perm true```
This may work around the issue, and afterwards you can set it back to false.

**Comment by MikeBauerCA** [comment]
Hi @Simnbf, thanks for using the Zowe CLI. For use in Jenkins, we recommend making use of [Jenkins secrets](https://www.jenkins.io/doc/book/using/using-credentials/) and passing them to the CLI via environmental vars. More information on where command arguments may come from is described [here](https://docs.zowe.org/stable/user-guide/cli-usingcli/#how-command-precedence-works) and detailed use of env vars is described [here](https://docs.zowe.org/stable/user-guide/cli-usingcli/#using-environment-variables).

I have also written a [blog](https://medium.com/zowe/continuous-integration-for-a-mainframe-app-800657e84e96) that includes an example. [Here is some sample code as well](https://github.com/zowe/zowe-cli-sample-scripts/blob/master/use_cases/CI_pipeline/Jenkinsfile#L38).

**Comment by awharn** [comment]
Hi @Simnbf. We did some investigation, and have determined that the root cause is most likely due to the absence of a prebuilt binary for keytar on the s390x architecture. Unfortunately, this means that the binary will need to be built from scratch. This should be handled at the plugin install time, but may fail if the following packages are not available on the system:

`python3` - version >=3.6.0
`make`
`gcc-c++`
`libsecret-devel`

I have opened a PR to document these requirements for zLinux.
